src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -2871,464 +3507,648 @@ public enum SupremeVictors implements LogicCardInfo {
+              }
+            }
           }
-          playRequirement{
+          onRemoveFromPlay{
+            eff.unregister()
           }
         };
       case CYNTHIA_S_GUIDANCE_136:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nLook at the top 7 cards of your deck, choose 1 of them, and put it into your hand. Put the other cards back on top of your deck. Shuffle your deck afterward."
+        return supporter (this) {
+          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card." +
+            "Look at the top 7 cards of your deck, choose 1 of them, and put it into your hand. Put the other cards back on top of your deck. Shuffle your deck afterward."
           onPlay {
+            my.deck.subList(0,7).select("Choose a card to put into your hand").moveTo(my.hand)
+            shuffleDeck()
           }
           playRequirement{
+            assert my.deck : "Your deck is empty"
           }
         };
       case CYRUS_S_INITIATIVE_137:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nFlip 2 coins. If either of them is heads, look at your opponent’s hand. For each heads, choose 1 card from your opponent’s hand and put it on the bottom of your opponent’s deck in any order."
+        return supporter (this) {
+          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card." +
+            "Flip 2 coins. If either of them is heads, look at your opponent's hand. For each heads, choose 1 card from your opponent's hand and put it on the bottom of your opponent's deck in any order."
           onPlay {
+            def count
+            flip 2, {
+              count ++
+            }
+            if(count) {
+              rearrange(opp.hand.select(count:count,"Choose $count cards to put on the botto of your opponent's deck")).showToOpponent("Cyrus's Initiative: Selected cards").moveTo(opp.deck)
+            }
           }
           playRequirement{
+            assert opp.hand : "Your opponent's hand is empty"
           }
         };
       case NIGHT_TELEPORTER_138:
-        return basicTrainer (this) {
+        return itemCard (this) {
           text "Flip a coin. If heads, put all cards in your hand on top of your deck. Then, search your deck for any 1 card and put it into your hand. Shuffle your deck afterward."
           onPlay {
+            flip {
+              my.hand.getExcludedList(thisCard).moveTo(hidden:true,my.deck)
+              my.deck.select("Search your deck for a card").moveTo(hidden:true,my.hand)
```

Comment:
Could you link the ruling here? (and no, not because of the memes)
https://compendium.pokegym.net/compendium-lvx.html#420